pkg/ruler/scheduler.go

```diff
@@ -134,17 +139,17 @@ func (s *scheduler) Run() {
 			if err != nil {
 				level.Warn(util.Logger).Log("msg", "scheduler: error updating configs", "err", err)
 			}
-		case <-s.stop:
+		case <-s.done:
 			ticker.Stop()
+			level.Debug(util.Logger).Log("msg", "scheduler config polling stopped")
 			return
 		}
 	}
 }
 
 func (s *scheduler) Stop() {
-	close(s.stop)
+	close(s.done)
 	s.q.Close()
-	<-s.done
```

Comment:
This double check is to give the scheduler time to shutdown.  Why remove it?

Reply:
It's not needed since the scheduler is gracefully shut down later by the ruler during the flush stage of the lifecycle. It empties the scheduling queue as a noop currently and once the scheduling_queue closes it's next channel or a set timeout is hit, it will complete the flush and finish shutting down.